### ISSUE TITLE
Regression in Performance between r0.9.0 and r0.10.0rc0.

### ISSUE THREAD

**Issue by cancan101** [tensorflow/tensorflow#3974]
### Environment info
Operating System:
```
Linux 4.4.11-23.53
```
### Observations

I am testing on a Tesla K80 (details below) using the following lines:
* `nvidia-docker run --rm -it -v /tmp/cifar10_data:/tmp/cifar10_data tensorflow/tensorflow:0.9.0-devel-gpu  bash -c 'ln -s /usr/local/nvidia/lib64/libcuda.so.1 /usr/lib/x86_64-linux-gnu/libcuda.so && python /usr/local/lib/python2.7/dist-packages/tensorflow/models/image/cifar10/cifar10_multi_gpu_train.py'`
* `nvidia-docker run --rm -it -v /tmp/cifar10_data:/tmp/cifar10_data tensorflow/tensorflow:0.10.0rc0-devel-gpu python /usr/local/lib/python2.7/dist-packages/tensorflow/models/image/cifar10/cifar10_multi_gpu_train.py`

On r0.0.9, I get: ~`575 examples/sec`.
On r0.10.0rc0 I get ~`425 examples/sec`.

Here are the device stats:
```
major: 3 minor: 7 memoryClockRate (GHz) 0.8235
pciBusID 0000:00:04.0
Total memory: 11.17GiB
Free memory: 11.11GiB
```

I have the: Nvidia Driver Version: 367.35

**Comment by vrv** [comment]
(Can you try the nightly pip at HEAD?  it's probably related to an eigen regression that we're in the process of cherry-picking into 0.10.0)

**Comment by cancan101** [comment]
It looks like tf nightly images are not getting built for `nightly-devel-gpu`: https://hub.docker.com/r/tensorflow/tensorflow/tags/

**Comment by vrv** [comment]
Indeed.  Can you try a non-docker version here: https://ci.tensorflow.org/view/Nightly/job/nigntly-matrix-linux-gpu/TF_BUILD_CONTAINER_TYPE=GPU,TF_BUILD_IS_OPT=OPT,TF_BUILD_IS_PIP=PIP,TF_BUILD_PYTHON_VERSION=PYTHON2,label=gpu-linux/ ?

cc @jendap and @gunan since I don't know how our dockers are built or pushed

**Comment by prb12** [comment]
Probably related to #3603 ?

**Comment by gunan** [comment]
@caisq has recently set up nightly-docker-* jobs to build and push nightly docker images for all our builds.
However, that job seems to be failing for a while now.
We will look into the docker failure.

I also cherrypicked Benoit's fix yesterday into r0.10. 
Once we fix the jobs that build docker images, I will ping this issue to try again.
I am assigning this to myself for now.

**Comment by cancan101** [comment]
This might be related to the nightlies failing: https://github.com/tensorflow/tensorflow/issues/3989.

I think the issue is cudnn 5.1 but am investigating.

**Comment by flx42** [comment]
@cancan101 This seems to be fixed in the nightly actually. This is the performance I got on a Titan X:
tensorflow/tensorflow:0.9.0-devel-gpu:
1497.08

tensorflow/tensorflow:0.10.0rc0-devel-gpu:
600.573

tensorflow/tensorflow:nightly-devel-gpu:
1934.54

**Comment by cancan101** [comment]
Are you able to get the Dockerfile on master to build?

**Comment by flx42** [comment]
@cancan101 nope, I'm getting the same error as you.

**Comment by flx42** [comment]
Well actually the fix was easy, see #3989

**Comment by gunan** [comment]
Closing this issue since after the fix is applied, the performance regression is gone.
Please reopen if I am wrong.
